code/modules/fishing/sources/_fish_source.dm

```diff
@@ -463,6 +466,24 @@ GLOBAL_LIST_INIT(specific_fish_icons, generate_specific_fish_icons())
 		if(severity >= EXPLODE_DEVASTATE)
 			reward.ex_act(EXPLODE_LIGHT)
 
+///Called when releasing a fish in a fishing spot with the TRAIT_CATCH_AND_RELEASE trait.
+/datum/fish_source/proc/readd_fish(obj/item/fish/fish, mob/living/releaser)
+	var/is_morbid = HAS_MIND_TRAIT(releaser, TRAIT_MORBID)
+	var/is_naive = HAS_MIND_TRAIT(releaser, TRAIT_NAIVE)
+	if(fish.status == FISH_DEAD) //ded fish won't repopulate the sea.
+		if(is_naive || is_morbid)
+			releaser.add_mood_event("fish_released", /datum/mood_event/fish_released, is_morbid && !is_naive, fish)
+		return
+	if((fish.type in fish_table) != is_morbid || is_naive)
```

Comment:
```suggestion
	if((fish.type in fish_table) != (is_morbid || is_naive))
```

Reply:
```suggestion
	if(((fish.type in fish_table) != is_morbid) || is_naive)
```
Being naive should always add the moodlet (sleepy is just eepy clown mentality)